packages/web-app/src/containers/transactionModals/DepositModal.tsx

```diff
@@ -81,28 +91,34 @@ const DepositModal: React.FC = () => {
       const amount = Number(input.amount) * Math.pow(10, tokenInfo.decimals);
       const allowance = await tokenAllowance(input.tokenAddress);
       if (allowance < amount) {
-        setDepositProcessState(TransactionState.APPROVING);
-        approve(input.tokenAddress, amount); // in promise
-        setDepositProcessState(TransactionState.LOADING);
+        try {
+          await approve(input.tokenAddress, amount);
+          setDepositProcessState(TransactionState.APPROVING);
```

Comment:
I think this should go before approved, and then once approved move to deposit right?